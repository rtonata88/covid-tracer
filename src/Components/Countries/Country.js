import React from 'react';

const Country = () => {
  return <div>Country Item</div>;
};
