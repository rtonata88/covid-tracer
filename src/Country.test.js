import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Country from './Components/Countries/Country';

const countries = [
  {
    id: 'afghanistan',
    name: 'Afghanistan',
    today_confirmed: 4,
  },
  {
    id: 'angola',
    name: 'Angola',
    today_confirmed: 4,
  },
];

const initialState = { countries: [] };
const mockStore = configureStore();
let store;

it('renders correctly', () => {
  store = mockStore(initialState);

  const countryData = countries.map((country, index) => (
    <Provider store={store} key={country.id}>
      <Country key={country.id} index={index} country={country} />
    </Provider>
  ));

  const tree = renderer.create(countryData).toJSON();
  expect(tree).toMatchSnapshot();
});
