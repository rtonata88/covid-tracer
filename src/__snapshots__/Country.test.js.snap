// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="bg-pink-1 col-sm-6 country-container"
    style={
      Object {
        "backgroundImage": "url(null)",
        "backgroundSizen": "cover",
      }
    }
  >
    <a
      className="country-block"
      href="/afghanistan"
      onClick={[Function]}
    >
      <div>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="country-name"
      >
        Afghanistan
        <p>
          <small>
            4
          </small>
        </p>
      </div>
    </a>
  </div>,
  <div
    className="bg-pink col-sm-6 country-container"
    style={
      Object {
        "backgroundImage": "url(null)",
        "backgroundSizen": "cover",
      }
    }
  >
    <a
      className="country-block"
      href="/angola"
      onClick={[Function]}
    >
      <div>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="country-name"
      >
        Angola
        <p>
          <small>
            4
          </small>
        </p>
      </div>
    </a>
  </div>,
]
`;
